crypto/txscript/src/opcodes/mod.rs

```diff
@@ -884,25 +873,125 @@ opcode_list! {
+    opcode OpTxInputCount<0xb3, 1>(self, vm) {
+        if vm.kip10_enabled {
+            match vm.script_source {
+                ScriptSource::TxInput{tx, ..} => {
+                    push_number(tx.inputs().len() as i64, vm)
+                },
+                _ => Err(TxScriptError::InvalidSource("OpInputCount only applies to transaction inputs".to_string()))
+            }
+        } else {
+            Err(TxScriptError::InvalidOpcode(format!("{self:?}")))
+        }
+    }
+    opcode OpTxOutputCount<0xb4, 1>(self, vm) {
+        if vm.kip10_enabled {
+            match vm.script_source {
+                ScriptSource::TxInput{tx, ..} => {
+                    push_number(tx.outputs().len() as i64, vm)
+                },
+                _ => Err(TxScriptError::InvalidSource("OpOutputCount only applies to transaction inputs".to_string()))
+            }
+        } else {
+            Err(TxScriptError::InvalidOpcode(format!("{self:?}")))
+        }
+    }
+    opcode OpTxLockTime<0xb5, 1>(self, vm) Err(TxScriptError::OpcodeReserved(format!("{self:?}")))
+    opcode OpTxSubnetId<0xb6, 1>(self, vm) Err(TxScriptError::OpcodeReserved(format!("{self:?}")))
+    opcode OpTxGas<0xb7, 1>(self, vm) Err(TxScriptError::OpcodeReserved(format!("{self:?}")))
+    opcode OpTxPayload<0xb8, 1>(self, vm) Err(TxScriptError::OpcodeReserved(format!("{self:?}")))
+    // Input related opcodes (following TransactionInput struct field order)
+    opcode OpTxInputIndex<0xb9, 1>(self, vm) {
+        if vm.kip10_enabled {
+            match vm.script_source {
+                ScriptSource::TxInput{idx, ..} => {
+                    push_number(idx as i64, vm)
+                },
+                _ => Err(TxScriptError::InvalidSource("OpInputIndex only applies to transaction inputs".to_string()))
+            }
+        } else {
+            Err(TxScriptError::InvalidOpcode(format!("{self:?}")))
+        }
+    }
+    opcode OpOutpointTxId<0xba, 1>(self, vm) Err(TxScriptError::OpcodeReserved(format!("{self:?}")))
+    opcode OpOutpointIndex<0xbb, 1>(self, vm) Err(TxScriptError::OpcodeReserved(format!("{self:?}")))
+    opcode OpTxInputScriptSig<0xbc, 1>(self, vm) Err(TxScriptError::OpcodeReserved(format!("{self:?}")))
+    opcode OpTxInputSeq<0xbd, 1>(self, vm) Err(TxScriptError::OpcodeReserved(format!("{self:?}")))
+    // UTXO related opcodes (following UtxoEntry struct field order)
+    opcode OpTxInputAmount<0xbe, 1>(self, vm) {
+        if vm.kip10_enabled {
+            match vm.script_source {
+                ScriptSource::TxInput{tx, ..} => {
+                    let [idx]: [i32; 1] = vm.dstack.pop_items()?;
+                    let utxo = usize::try_from(idx).ok()
+                        .and_then(|idx| tx.utxo(idx))
+                        .ok_or_else(|| TxScriptError::InvalidInputIndex(idx, tx.inputs().len()))?;
+                    push_number(utxo.amount as i64, vm)
```

Comment:
Although KAS value is bounded below `i64::MAX`, I think the implementation will be more robust if we script err on overflow here 

```suggestion
                    push_number(utxo.amount.try_into().map_err(|e: TryFromIntError| TxScriptError::NumberTooBig(e.to_string()))?, vm)
```